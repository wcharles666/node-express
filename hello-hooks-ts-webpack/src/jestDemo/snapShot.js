export const genneratorConfig = () => {
  return {
    server: 'http://wwww.baidu.com',
    port: 8080,
    time: new Date(),
  };
};

export const genneratorConfig1 = () => {
  return {
    server: 'http://wwww.baidu.com',
    port: 8080,
    time: new Date(),
  };
};
