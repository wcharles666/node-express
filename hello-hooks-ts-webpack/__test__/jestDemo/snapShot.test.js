import { genneratorConfig, genneratorConfig1 } from '../../src/jestDemo/snapShot';

// 普通写法
// test('测试配置文件快照 ', () => {
//   expect(genneratorConfig()).toEqual({
//     server: 'http://wwww.baidu.com',
//     port: 8080,
//   });
// });

// 快照写法
test('测试配置文件快照1snap ', () => {
  expect(genneratorConfig()).toMatchSnapshot({
    time: expect.any(Date),
  });
});

test('测试配置文件快照2snap ', () => {
  expect(genneratorConfig1()).toMatchSnapshot({
    time: expect.any(Date),
  });
});

test('测试配置文件快照2snap 显示在行内测试用例文件中 ', () => {
  expect(genneratorConfig1()).toMatchInlineSnapshot(
    {
      time: expect.any(Date),
    },
    `
    Object {
      "port": 8080,
      "server": "http://wwww.baidu.com",
      "time": Any<Date>,
    }
  `
  );
});
