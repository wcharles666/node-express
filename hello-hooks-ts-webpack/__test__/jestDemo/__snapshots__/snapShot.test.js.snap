// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试配置文件快照1snap  1`] = `
Object {
  "port": 8080,
  "server": "http://wwww.baidu.com",
  "time": Any<Date>,
}
`;

exports[`测试配置文件快照2snap  1`] = `
Object {
  "port": 8080,
  "server": "http://wwww.baidu.com",
  "time": Any<Date>,
}
`;
